lib/galaxy/datatypes/images.py

```diff
@@ -285,6 +287,36 @@ def sniff_prefix(self, file_prefix):
         return False
 
 
+class Mrc2014(Binary):
+    """
+    MRC/CCP4 2014 file format (.mrc).
+    https://www.ccpem.ac.uk/mrc_format/mrc2014.php
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('1.mrc')
+    >>> Mrc2014().sniff(fname)
+    True
+    >>> fname = get_test_fname('2.txt')
+    >>> Mrc2014().sniff(fname)
+    False
+    """
+    file_ext = 'mrc'
+
+    def sniff(self, filename):
+        # Handle the wierdness of mrcfile:
+        # https://github.com/ccpem/mrcfile/blob/master/mrcfile/validator.py#L88
+        with contextlib.redirect_stdout(None):
```

Comment:
Nice, I didn't know this one

Reply:
Should we worry about this caveat? "Note that the global side effect on `sys.stdout` means that this context manager is not suitable for use in library code and most threaded applications. It also has no effect on the output of subprocesses. However, it is still a useful approach for many utility scripts."

Reply:
Hmm good point, `global` and `most threaded applications` is as vague as it gets. Probably safer to pass something else in the print_file argument.

Reply:
Should it be an open tmp file or something else?

Reply:
StringIO is probably fine.